src/switchroot/ostree-remount.c

```diff
@@ -133,10 +139,55 @@ relabel_dir_for_upper (const char *upper_path, const char *real_path, gboolean i
 #endif
 }
 
+// ostree-prepare-root sets things up so that /sysroot points to the "physical" (real) root in the
+// initramfs, and then with composefs `/` is an overlay+EROFS that holds references to content in
+// that physical filesystem.
+//
+// In a typical mutable system where the OS is in a mutable `/` (or `/usr), systemd explicitly
+// skips unmounting both `/` and `/usr`. It will remount them read-only though - and that's
+// the semantic we want to match here.
+static void
+do_shutdown (void)
+{
+  // We expect this to be read-only by default on most modern systems, but
+  // in case it's not, make it read-only now.
+  do_remount ("/sysroot", false);
```

Comment:
https://github.com/ostreedev/ostree/blob/a5a52e01edd565cca368b946d5e5e4a333b3f350/src/switchroot/ostree-remount.c#L61-L63
We want to unconditionally remount read-only, because the bind mount on top might be read only,
but not the original mount point under it.
In my tests using `mount -o remount,ro` works, so we need a `do_remount_force`